src/LatteContext/CollectedData/CollectedVariable.php

```diff
@@ -63,12 +63,12 @@ public function isDeclared(): bool
         return $this->declared;
     }
 
-    public static function build(Node $node, Scope $scope, string $name, Type $type, bool $declared = false): self
+    public static function build(Node $node, Scope $scope, string $name, Type $type, bool $declared = false, bool $mightBeUndefined = false): self
```

Comment:
@MartinMystikJonas do you remember what is $declared for? There is no doc for that

Reply:
It is used to differentiate between discovered types and explicitly declared types (using @phpstan-latte-* annotations) whoch have higher priority when types are conflicting

Reply:
Oh, I see. OK, so I'm not duplicating functionality here